### PR TITLE
Remove unused c10/util/C++17.h inclusion and outdated checks

### PR DESCRIPTION
**Pull request by cyyever**
This is a continued work to clean up pre-C++17 code.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: c10/util/C++17.h [hunk 1]
```diff
@@ -8,17 +8,6 @@
...
-    __GNUC__ < 5
-#error \
-    "You're trying to build PyTorch with a too old version of GCC. We need GCC 5 or later."
-#endif
-
-#if defined(__clang__) && __clang_major__ < 4
-#error \
-    "You're trying to build PyTorch with a too old version of Clang. We need Clang 4 or later."
-#endif
-
```
Comment (malfet) [inline c10/util/C++17.h:21]: WE better keep those checks in place, otherwise one might run into https://github.com/pytorch/pytorch/issues/120020

**Pull request closed without merging**